.github/workflows/ci.yml

```
@@ -23,7 +23,9 @@ jobs:
       - run: python -m flake8
       - run: python -m mypy fluent.syntax/fluent fluent.runtime/fluent
   test:
-    runs-on: ubuntu-latest
+    # Workaround Python 3.7 no longer supported https://github.com/actions/setup-python/issues/962#issuecomment-2414454450
+    #runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
     strategy:
       matrix:
         python-version: [3.7, 3.8, 3.9, "3.10", 3.11, 3.12, pypy3.9, pypy3.10]
```

If there are test failures on Windows, shouldn't we also run the test suite there?

```suggestion
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-22.04, windows-2022]
        python-version: [3.7, 3.8, 3.9, "3.10", 3.11, 3.12, pypy3.9, pypy3.10]
```

---

It feels okay, but if what you care about is that tests pass, maybe there should also be tests for `normalize_path` and related file simulating code. I can add them if you want them.

---

Sorry, I don't understand what you said (what code paths did you mean).

Anyway I decide to push my work to show it first (457d6980c8e26882da92f49d9c46d15fb4b3ab9e).

My point was that it was the file simulation code that caused the test errors on a different platform, so it makes sense to test the file simulation code.

It is an opinionated change to no longer reflect files on disk, because it seems unnecessary and unused* in current tests, and may make it difficult when you want to simulate some files don't exist and there are real files which names collide with them. I'd be happy to know what you think.

\* It can be added if it is needed later

---

I'll mark this change request conversation as resolved, since I already applied the suggestions, let's continue the discussion outside. Sorry about being clumsy, this is my first PR.